### PR TITLE
DRILL-6793: FragmentExecutor cannot send its final state for the case…

### PR DESCRIPTION
**Pull request by KazydubB**
… when RootExec root wasn't initialized

### CODE REVIEW

File: contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java [hunk 1]
```diff
@@ -530,8 +530,10 @@ public void close() {
 
   @Override
   public String toString() {
-    StringBuilder sb = new StringBuilder("MaprDBJsonRecordReader[Table=")
-        .append(table.getPath());
+    StringBuilder sb = new StringBuilder("MaprDBJsonRecordReader[Table=");
+    if (table != null) {
```
Comment (vvysotskyi) [inline contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java]: For the case when `table` is null, the string will be formed incorrectly - `Table=` should be appended only if `table != null`

File: contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java [hunk 2]
```diff
@@ -530,8 +530,10 @@ public void close() {
...
+      sb.append(table.getPath());
+    }
     if (reader != null) {
       sb.append(", Document ID=")
```
Comment (vvysotskyi) [inline contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java]: Also, please move `, `string here and below to the previous appenders.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroRecordReader.java [hunk 3]
```diff
@@ -403,7 +403,9 @@ public void close() {
   public String toString() {
     long currentPosition = -1L;
     try {
-      currentPosition = reader.tell();
+      if (reader != null) {
+        currentPosition = reader.tell();
+      }
     } catch (IOException e) {
       logger.trace("Unable to obtain reader position: " + e.getMessage());
```
Comment (vvysotskyi) [inline exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroRecordReader.java]: Please log the stack trace of the exception.

File: contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java (hunk 1 above)
Comment (KazydubB) [reply contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java]: Thanks for pointing it out, done.

File: contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java (hunk 2 above)
Comment (KazydubB) [reply contrib/format-maprdb/src/main/java/org/apache/drill/exec/store/mapr/db/json/MaprDBJsonRecordReader.java]: Made previous field to be present in any case. It seems there is no need in doing so.

File: exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroRecordReader.java (hunk 3 above)
Comment (KazydubB) [reply exec/java-exec/src/main/java/org/apache/drill/exec/store/avro/AvroRecordReader.java]: Done.

**Pull request closed without merging**